configs/default/SPRO-SPRACINGH7EXTREME.config

```diff
@@ -0,0 +1,225 @@
+# Betaflight / STM32H750 (SP7E) 4.4.0 Dec 26 2022 / 12:28:11 (7671c7eb01) MSP API: 1.45
+
+#define TARGET_BOARD_IDENTIFIER "SP7E"
+#define USBD_PRODUCT_STRING "SPRacingH7EXTREME"
+
+#
+# Settings required to boot
+#
+# Note: hardware for all possible config storage subsystems needs to be known and initialised BEFORE loading
+# a config from it can work. (SPI/QSPI/SDCARD).
+#
+
+#define EEPROM_SIZE=8192
+
+#define USE_SPRACING_PERSISTENT_RTC_WORKAROUND
+
+#define USE_BUTTONS
```

Comment:
What are the buttons for? should they be using resource mapping?

Reply:
buttons are used for various pre-config-loaded activities such as resetting the config, USB MSC and also for VTX band/channel/power settings.  See the main BF codebase: `init.c`, `vtx_control.c`, `pg\usb.c`, etc.

Reply:
So the point is still valid, these should be a user option gate. So the user can chose to enable them, in the build, and then apply resource mapping to configure them.

Reply:
can you explain what you mean by 'user gate' and how to create one.  I don't recall seeing any documentation or use of that term until now.

If the user wants to enable buttons then the expectation would be that they always work, and not 'only work after config is loaded' so with that in mind they /have/ to be compile-time options in the case of the H750 targets.

Reply:
USE_PINIO is an example of such an option / gate. The user can choose now - with cloud build - to have it included in the firmware or not.

Essentially users now have both compile time and runtime options, whereas before they only had runtime.

Reply:
the problem is users don't know what hardware engineers and target authors intended and or how the circuits work.  it's often not OK to just leave circuits at some random default or the pin be re-assigned accidentally to another function by a user, hence the requirement for a base target on which users can build upon, but not remove from.

For this specific issue buttons are a hardware feature which requires correct initialization prior to a config to work.

Things like a GPS support, or some legacy RX protocol that no-one needs are perfect examples of what can be compiles out, but not things like `USE_PID_AUDIO` and `USE_BUTTONS`.

Reply:
It's your target so you can force it if you want.